vespalib/src/vespa/vespalib/datastore/bufferstate.h

```diff
@@ -130,7 +130,27 @@ class BufferState
     BufferTypeBase *getTypeHandler() { return _typeHandler.load(std::memory_order_relaxed); }
 
     void resume_primary_buffer(uint32_t buffer_id);
+};
 
+class BufferAndMeta {
```

Comment:
Consider moving this class to a separate header file.

Reply:
I prefer keeping it here as it is a small subset of the BufferState.

Reply:
👍